compiler/rustc_codegen_rmc/src/compiler_interface.rs

```diff
@@ -164,13 +160,13 @@ impl CodegenBackend for GotocCodegenBackend {
 
             let symtab_filename = base_filename.with_extension("symtab.json");
             debug!("output to {:?}", symtab_filename);
-            let mut out_file = ::std::fs::File::create(&symtab_filename).unwrap();
-            write!(out_file, "{}", result.symtab.to_irep().to_json().pretty().to_string()).unwrap();
+            let out_file = ::std::fs::File::create(&symtab_filename).unwrap();
+            serde_json::to_writer(out_file, &result.symtab.to_irep()).unwrap();
```

Comment:
Wrap out_file with a BufWriter and see if that cures your performance problems.

Reply:
That's a good idea. Let me try.